### PR TITLE
Add github actions ci build

### PR DESCRIPTION
**Pull request by srose**

### CODE REVIEW

**Comment by srose** [comment]
docker-compose is used to setup the stage, e2e cypress tests are executed via docker run.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: ACME Maven Build
```
Comment (thomasdarimont) [inline .github/workflows/build.yml:1]: ACME Keycloak Build

File: .github/workflows/e2e-tests.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
+name: Acme e2e Test CI
+
+on:
```
Comment (thomasdarimont) [inline .github/workflows/e2e-tests.yml:3]: Can we run this for every pull request?

File: .github/workflows/e2e-tests.yml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        with:
+          node-version: '14.x'
+      - name: Prepare artifacts (e.g. extensions) for docker-compose stack
+        run: mvn --batch-mode -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn -DskipTests verify --file pom.xml
+      - name: Build the docker-compose stack
+        run: |
+          export USER=`id -u`
+          export GROUP=
+          echo "Running as ${USER}:${GROUP}"
+          java bin/prerequisites.java
```
Comment (thomasdarimont) [inline .github/workflows/e2e-tests.yml:29]: Perhaps we can rename `prerequisitis.java` to something like `envcheck.java` as it is simpler?

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 11
```
Comment (thomasdarimont) [inline .github/workflows/build.yml:14]: Please add an empty newline for clarity between steps

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+      - name: Maven - verify
+        run:  mvn --batch-mode -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn verify --file pom.xml
+      - name: Maven - integration-test
+        run: mvn --batch-mode -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn -Pwith-integration-tests test --file pom.xml
+      - name: Maven - build image
+        run: mvn --batch-mode -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn -DskipTests io.fabric8:docker-maven-plugin:build --file pom.xml
```
Comment (thomasdarimont) [inline .github/workflows/build.yml:29]: Could you also run trivy image scan here?

**Review by thomasdarimont — CHANGES_REQUESTED** [review]
Looks good! Just some minor comments in between.

File: .github/workflows/build.yml (hunk 5 above)
Comment (srose) [reply .github/workflows/build.yml:29]: Good idea, will do

Comment (srose) [reply .github/workflows/build.yml:29]: Publishing also the test report from cypress is something comes to my mind through that comment :)

File: .github/workflows/e2e-tests.yml (hunk 2 above)
Comment (srose) [reply .github/workflows/e2e-tests.yml:3]: Should already be in: this is how it looks like once pulled in:
https://github.com/srose/keycloak-project-example/pull/3/checks

**Comment by srose** [comment]
Thanks for the review, https://github.com/srose/keycloak-project-example/pull/5/checks the status is visible once activated.

File: .github/workflows/e2e-tests.yml [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        run: |
+          docker ps -a
+          docker inspect dev_acme-keycloak_1
+          docker logs --details dev_acme-keycloak_1
+
+      - name: Run cypress tests
+        working-directory: ./keycloak/e2e-tests
+        run: |
+          yarn install
+          docker run --network container:dev_acme-keycloak_1 --rm -v ${PWD}:/e2e -w /e2e --entrypoint=cypress cypress/included:5.6.0 run
```
Comment (thomasdarimont) [inline .github/workflows/e2e-tests.yml:55]: Align cypress version with e2e tests

Comment (srose) [reply .github/workflows/e2e-tests.yml:55]: good catch

**Comment by thomasdarimont** [comment]
Thanks @srose let's give it a try.

**Pull request merged**